google/cloud/logging_v2/handlers/structured_log.py

```diff
@@ -84,3 +87,13 @@ def format(self, record):
         # convert to GCP structred logging format
         gcp_payload = self._gcp_formatter.format(record)
         return gcp_payload
+
+    def emit(self, record):
+        if google.cloud.logging_v2.instrumentation_emitted is False:
```

Comment:
I think I prefer to leave it explicit.  It seems a bit clearer than relying on an opaque return value, though I'd be interested in @daniel-sanche 's thoughts on which method seems more pythonic.

Reply:
The problem here might be potential race condition (despite a fact that Python usually used under GIL) - hence you want to minimize time between variable value is taken and set... Given a fact that this library can be used anywhere, it might come to a problem of multiple records written since variable is checked in one function and set in another.
Goof programming style in such case would be calling atomic "get and set" function, but apparently I am not sure one exists in Python - in this case you could make that "get and set" in `setInstrumentationEmittedState()` at least to keep good programming style

Reply:
Yeah I think in general the Python style prefers not to create boilerplate setters and getters when not necessary, but Alex makes a good point that there could be a race condition here, so we should make this atomic

Reply:
Ok.  I think I want to work on that name a bit. `setInstrumentationEmittedState()` implies that it's just setting the variable (presumably to true, with or without an explicit argument.) I'm not sure how best to capture that we're returning the previous state other than very clear documentation.  Maybe it's not important, but if we could make it clear in the name, that would be ideal.

Reply:
Yeah, make sure that it is documented at least